### PR TITLE
Adds Rails 5 (beta3) support.

### PR DESCRIPTION
**Pull request by jejacks0n**
There were a few minor things, but they are mostly limited to the controller, with a few minor adjustments to configuration.

### CODE REVIEW

File: app/controllers/teaspoon/suite_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class Teaspoon::SuiteController < ActionController::Base
-  before_filter :check_env
-  before_filter :prepend_fixture_paths
+  def self.before(*args)
+    respond_to?(:before_action) ? before_action(*args) : before_filter(*args)
```
Comment (jejacks0n) [inline app/controllers/teaspoon/suite_controller.rb]: Cheers hound.. sadly for rails 3 support it must stay. :+1:

**Comment by connorshea** [comment]
We're interested in this for GitLab, any updates?

**Comment by jejacks0n** [comment]
I'll look at it today @connorshea.

**Comment by connorshea** [comment]
Can the Rails 5 Gemfile be modified to just `rails < 5.1`, that way the rc, final, and patch (5.0.1) releases won't need gem updates? That'd be the ideal solution, although I can understand why you'd be wary to support versions you haven't tested yet.

**Comment by jejacks0n** [comment]
Your comment did remind me to lighten up the version in the gemfile for travis -- this is sort of how we figure out when a new version is breaking something.

**Comment by connorshea** [comment]
I believe there's an issue with Bundler/RubyGems where `~> 5.0` doesn't include prerelease versions. This may have been fixed since I last ran into it a few months ago, but if the build doesn't work that's probably why :)

`< 5.1` should work if that doesn't.

**Comment by jejacks0n** [comment]
I don't think the build will work anyway, haha, but yeah, I wasn't sure about that and figured I'd find out. =)

**Comment by jejacks0n** [comment]
@connorshea how likely is it that you're able to test out master and make sure things are good before we release 1.1.5?

**Comment by connorshea** [comment]
Give me two hours and I can try it out on GitLab, mention me again if I haven't commented by late tonight/tomorrow morning.

**Comment by connorshea** [comment]
Should also note that we're not on Rails 5 yet, unfortunately. So we can't actually test with Rails 5 at the moment :/

I don't work on any other projects that are using Rails 5 and Teaspoon, so I'm not sure how useful my testing would be.

**Comment by jejacks0n** [comment]
No worries, and thanks. There's higher value in making sure that it's working correctly in a large project with Rails 4 first -- Rails 5 fixes can roll out as they solidify that release. Follow up question for you -- why track this PR if you didn't directly need Rails 5 support yet? =) Just curious about how you're handling that process generally.

**Comment by connorshea** [comment]
It's partly for us and partly for the good of the community overall :smiley:

So far I've found two gems that seem to be abandoned, and have commented on a few PRs/issues for a handful of others. I didn't want to try upgrading once Rails 5 proper comes out and be blindsided if we find that we have 6 gems that haven't had compatibility releases yet. 

I have a full list of all the gems that need updating [in this comment thread](https://gitlab.com/gitlab-org/gitlab-ce/issues/14286#note_4272270). I've been doing a pretty comprehensive sweep of our Gemfile.lock for the last day or so to see if the gems that are locked to 4.x versions have had compatibility releases yet. Generally gems are updated one-at-a-time, so I've had to open a new Merge Request for every gem I've updated. Bit of a time consuming process, but better to do it now than deal with a dozen Bundler dependency conflicts when we're actually updating :D

tl;dr I do it because I'm crazy

**Comment by connorshea** [comment]
Anyway, [here's the Teaspoon build for GitLab when using Teaspoon's master branch](https://gitlab.com/connorshea/gitlab-ce/builds/937364).

**Comment by jejacks0n** [comment]
Cheers, and yes, you have more heart than myself. =) It's been hard enough to keep one gem updated promptly with Rails+Sprockets over the years. Will release a new version shortly, so you can probably update later today and be good rolling forward (unless there's a big change between now and final release of Rails 5).

**Comment by connorshea** [comment]
Excellent, thank you very much for the help and for maintaining the gem :D
